### PR TITLE
some improvements to the llvm/codegen makefiles

### PR DESCRIPTION
**Pull request by vtjnash**

### CODE REVIEW

File: deps/llvm.mk [hunk 1]
```diff
@@ -173,7 +173,7 @@ endif
 
 ifneq ($(LLVM_VER),svn)
 ifeq (,$(findstring rc,$(LLVM_VER)))
-ifeq ($(shell [ $(LLVM_VER_MAJ) -ge 8 -a $(LLVM_VER) != 8.0.0 ]; echo $$?),0)
+ifeq ($(shell [ x"$(LLVM_VER)" = x"8.0.1" ]; echo $$?),0)
```
Comment (Keno) [inline deps/llvm.mk]: Won't this conditional fail for LLVM 9 now?

Comment (vtjnash) [reply deps/llvm.mk]: yeah, turns out llvm 8.0.1 was the only release they decided not to upload to releases.llvm.org and use a github url instead. I mistakenly assumed that mean they were switching to that.

Comment (Keno) [reply deps/llvm.mk]: :/

File: test/llvmpasses/Makefile [hunk 2]
```diff
@@ -1,8 +1,14 @@
 SRCDIR := $(abspath $(dir $(lastword $(MAKEFILE_LIST))))
 JULIAHOME := $(abspath $(SRCDIR)/../..)
 include $(JULIAHOME)/Make.inc
-test:
+
+check: $(SRCDIR)
```
Comment (Keno) [inline test/llvmpasses/Makefile]: update the CI configuration that runs these tests to use the `check` target instead?

Comment (vtjnash) [reply test/llvmpasses/Makefile]: We don't seem to be hardcoding the name anywhere. Do you know of a case?

Comment (Keno) [reply test/llvmpasses/Makefile]: Fair, I guess this is the default target, so things will keep working.

**Review by Keno — APPROVED** [review]

**Pull request merged**